src/wp-includes/block-editor.php

```diff
@@ -308,36 +308,37 @@ function get_block_editor_settings( array $custom_settings, $block_editor_contex
 	);
 
 	$global_styles = array();
-	$presets       = array(
-		array(
-			'css'            => 'variables',
-			'__unstableType' => 'presets',
-		),
-		array(
-			'css'            => 'presets',
-			'__unstableType' => 'presets',
-		),
-	);
-	foreach ( $presets as $preset_style ) {
-		$actual_css = wp_get_global_stylesheet( array( $preset_style['css'] ) );
-		if ( '' !== $actual_css ) {
-			$preset_style['css'] = $actual_css;
-			$global_styles[]     = $preset_style;
+	if ( ! empty( $block_editor_context->post ) ) {
```

Comment:
For me the ideal solution is to load these in all editors (no exceptions) but have a flag to know which ones are generated (global styles) and which ones are not.

Reply:
Looking at the other PR https://github.com/WordPress/gutenberg/pull/40185 we do need two orthogonal things:

- be able to tell which are block styles and which are preset styles (the later should still be loaded when the user disables the "theme styles")
- be able to tell which styles come from "theme.json" and which don't (so we can overwrite the former in the site editor)

Load them all for all editors but mark them with the "styles source" solves the issue as well. Going to refactor this.

Reply:
https://github.com/WordPress/wordpress-develop/pull/2542/commits/175b959c0896cf5a2328ccea56fbce2a37ef5628 updates the categories to be:

- Types: "presets" vs "styles" (was "theme" before)
- Sources: "theme_json" vs "css"

Reply:
This works for me 👍 I agree we need both things. Good things these are still unstable.

Do you think there's a way to detect the right type and source for how things were in WP 5.9 (mainly thinking about Gutenberg plugin compatibility here).

Reply:
I wonder if `theme_json` is a good value there. Technically it can come from variations, user global styles. so maybe "global styles" is the right value there, or something like it?